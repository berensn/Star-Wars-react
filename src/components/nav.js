import React, { Component } from 'react';
import {NavLink} from 'react-router-dom';
import {
  NavItem,
  NavElements,
  NavLinq
} from '../styles/nav.style';
import '../App.css'

class Nav extends Component {  
  render(){
    var people = 'people';
    var planets = 'planets';
    var species = 'species';
    var vehicles = 'vehicles';
    var starships = 'starships';
    var films = 'films';
    return(
      <div>
        <NavElements>
          <NavItem><NavLinq to='/home' >Home</NavLinq></NavItem>
          <NavItem><NavLinq to={`/category/${people}/`} >People</NavLinq></NavItem> 
          <NavItem><NavLinq to={`/category/${planets}/`} >Planets</NavLinq></NavItem> 
          <NavItem><NavLinq to={`/category/${species}/`} >Species</NavLinq></NavItem> 
          <NavItem><NavLinq to={`/category/${vehicles}/`} >Vehicles</NavLinq></NavItem> 
          <NavItem><NavLinq to={`/category/${starships}/`} >Starships</NavLinq></NavItem> 
          <NavItem><NavLinq to={`/category/${films}/`} >Films</NavLinq></NavItem>
         </NavElements>
      </div>
    );
  }
}

export default Nav;