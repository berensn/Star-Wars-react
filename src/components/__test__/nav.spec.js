import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../nav.js';
import Category from '../category';
import { Route } from 'react-router-dom';
import { NavLinq } from '../../styles/nav.style';

describe('<Nav />', () => {
  it('should include the correct links', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Nav />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match rendered link format', () => {
    const tree = renderer
      .create(
      <MemoryRouter>
        <NavLinq to='/component/people'>People</NavLinq>
      </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the correct route format', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Route path='category/:name' component={Category} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});