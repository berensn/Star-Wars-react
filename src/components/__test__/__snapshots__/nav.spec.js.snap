// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should include the correct links 1`] = `
<div>
  <nav
    className="sc-htpNat hGxvve"
  >
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/home"
        onClick={[Function]}
        style={undefined}
      >
        Home
      </a>
    </span>
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/category/people/"
        onClick={[Function]}
        style={undefined}
      >
        People
      </a>
    </span>
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/category/planets/"
        onClick={[Function]}
        style={undefined}
      >
        Planets
      </a>
    </span>
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/category/species/"
        onClick={[Function]}
        style={undefined}
      >
        Species
      </a>
    </span>
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/category/vehicles/"
        onClick={[Function]}
        style={undefined}
      >
        Vehicles
      </a>
    </span>
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/category/starships/"
        onClick={[Function]}
        style={undefined}
      >
        Starships
      </a>
    </span>
    <span
      className="sc-bwzfXH dSnPBo"
    >
      <a
        aria-current={null}
        className="sc-bdVaJa lkrUhG"
        href="/category/films/"
        onClick={[Function]}
        style={undefined}
      >
        Films
      </a>
    </span>
  </nav>
</div>
`;

exports[`<Nav /> should match rendered link format 1`] = `
<a
  aria-current={null}
  className="sc-bdVaJa lkrUhG"
  href="/component/people"
  onClick={[Function]}
  style={undefined}
>
  People
</a>
`;

exports[`<Nav /> should render the correct route format 1`] = `null`;
